vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -51,6 +48,18 @@ class _BaseAction(VizroBaseModel):
     function: ClassVar[Callable[..., Any]]
     outputs: ClassVar[Union[list[str], dict[str, str]]]
 
+    # These are set in the make_actions_chain validator (same for both Action and _AbstractAction).
+    # In the future a user would probably be able to specify something here that would look up a key in
+    # _action_triggers or just a full _IdProperty. The _first_in_chain and _prevent_initial_call_of_guard would remain
+    # private though. These are required for correct functioning of the actions chain guard.
+    _trigger: _IdProperty = PrivateAttr()
```

Comment:
Key change: all actions now have `_trigger` property. This would look like e.g. `graph_id.cellClicked`.